Google signing for React Native apps

MonicaG opened the pull request:
This is a rewrite of the [Google Signing for React Native Application](https://github.com/bcgov/mobile-signing-service/wiki/Google-Signing-for-React-Native-Application).

I updated it to describe how to do the signing in a CI/CD pipeline, rather than from a developer's desktop.

I removed the steps on how to navigate in Google Play Console. I provided a link to the google documentation instead.

kaitlyn-rosenburg commented on docs/google_react_native_signing.md:
```diff
@@ -1 +1,147 @@
+# Google signing for React Native apps
+
+
+Refer to React Native's [Publishing to Google Play Store](https://reactnative.dev/docs/signed-apk-android) guide for an overview of app signing. That guide describes how to sign an app from a developer's computer. This page describes how to setup those steps in a CI/CD pipeline using GitHub Actions.
```
```suggestion
Learn how to setup Google signing in a CI/CD pipeline using GitHub Actions.

Need a general overview of what app signing is? Review React Native's [Publishing to Google Play Store](https://reactnative.dev/docs/signed-apk-android) guide which describes how to sign an app from a developer's computer. 
```

kaitlyn-rosenburg replied:
Always start the page with an intro sentence on what the content is about

kaitlyn-rosenburg commented on docs/google_react_native_signing.md:
```diff
@@ -1 +1,147 @@
+
+Refer to React Native's [Publishing to Google Play Store](https://reactnative.dev/docs/signed-apk-android) guide for an overview of app signing. That guide describes how to sign an app from a developer's computer. This page describes how to setup those steps in a CI/CD pipeline using GitHub Actions.
+
+## Prerequisite
```
It would be nice to see this page restructured to use steps/numbers to walk someone through what they need to do.

MonicaG replied:
I added the steps at the top of the page, with links to the sections. I wasn't sure how to restructure it into steps, as each step had a fair bit of detail.

kaitlyn-rosenburg commented on docs/google_react_native_signing.md:
```diff
@@ -1 +1,147 @@
+
+## Setup GitHub secrets
+
+In your app's GitHub repo, setup the following secrets:
```
```suggestion
In your app's GitHub repo, setup 4 secrets:
```

kaitlyn-rosenburg commented on docs/google_react_native_signing.md:
```diff
@@ -1 +1,147 @@
+* ANDROID_SIGNING_STORE_PASSWORD={password-from-above-step}
+* ANDROID_SIGNING_KEY_PASSWORD={password-from-above-step}
+
+Use the following command to encode the keystore file as base64:
```
```suggestion
To encode the keystore file as base64, use this command:
```

kaitlyn-rosenburg commented on docs/google_react_native_signing.md:
````diff
@@ -1 +1,147 @@
+    }
+```
+
+Additionally, we need to tell Gradle to use our new config for release builds:
````
```suggestion
You also need to tell Gradle to use the new config for release builds:
```

kaitlyn-rosenburg commented on docs/google_react_native_signing.md:
```diff
@@ -1 +1,147 @@
+
+## GitHub Action
+
+ See the [bc-mobile-wallet](https://github.com/bcgov/bc-wallet-mobile) project for a complete GitHub Action workflow. Below is a basic GitHub Action.
```
```suggestion
Review the [bc-mobile-wallet](https://github.com/bcgov/bc-wallet-mobile) project for a complete GitHub Action workflow. 

Example of a basic GitHub Action:
```

kaitlyn-rosenburg commented on docs/google_react_native_signing.md:
```diff
@@ -1 +1,147 @@
+## Upload the build
+
+### GitHub Action Upload
+See the [Publish Your App](publish.md) page for information on how to upload from your CI/CD pipeline.
```
```suggestion
Review the [Publish Your App](publish.md) page for information on how to upload from your CI/CD pipeline.
```

kaitlyn-rosenburg replied:
again, avoid use of "see"

MonicaG commented:
@kaitlyn-rosenburg - I've updated the page with your suggestions. You can [review it here](https://github.com/bcgov/mobile-developer-guide/pull/18/commits/fe95a87b9e8890a6b516886cc2fe3162d36fad1b). As noted above, I added in the steps at the top of the page with links to the details. I'm open to other ways of structuring the content. This was just my first attempt.
